### PR TITLE
Resolve "Introduce release command"

### LINKED ISSUE #30: Introduce release command
To create releases based on `semver`.

### PR DESCRIPTION
**Pull request by noxan**
Close #30

### CODE REVIEW

File: src/commands/release.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Arguments } from 'yargs';
+import * as git from 'simple-git/promise';
+import * as opn from 'opn';
+import * as readPkg from 'read-pkg';
```
Comment (noxan) [inline src/commands/release.ts]: Would a simple `import`/`require` do the job?

File: src/commands/release.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import * as semver from 'semver';
+import * as writePkg from 'write-pkg';
+
+const productionBranchname = 'production';
+
+const validateVersion = (version: string) =>
```
Comment (noxan) [inline src/commands/release.ts]: Is this function being used somewhere?

File: src/commands/release.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  !!semver.valid(version) || false;
+
+const releaseCommand = async (
+  command: any /* ask Richard about this */,
```
Comment (noxan) [inline src/commands/release.ts]: `command` is the command which is being selected. speaking of `release` in this case. what is the specific question?

File: src/commands/release.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  project: any,
+  args: Arguments,
+  authToken: string,
+) => {
+  const SEMVER_LEVELS = ['major', 'minor', 'patch'];
+  const repository = git(process.cwd());
+  const version = (await readPkg()).version;
+  const branchName = (await repository.status()).current;
+
+  let nextVersion = command;
```
Comment (noxan) [inline src/commands/release.ts]: This will result in `release` as version?

File: src/commands/release.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  // Checkout new branch `release-x.y.z`
+  const releaseBranch = `release-${nextVersion}`;
+  repository.checkoutLocalBranch(releaseBranch);
+
+  // Writes version number in package.json
+  writePkg({ version: nextVersion });
+
+  await repository.add('package.json');
+  // await repository.commit('package.json');
```
Comment (noxan) [inline src/commands/release.ts]: In addition to `commit` I think we also need to `push` the changes.

**Comment by noxan** [comment]
Oh it also removes all content except for `version` from `package.json`... 🤔

**Comment by noxan** [comment]
Just had a brief look but looks great 👍 

It just still adds some extra fields and makes some formatting changes to `package.json` file, which might be related to the `readPkg` function.

**Comment by lcsrinaldi** [comment]
@noxan yes we have to work more on this package.json solution, as a tool user I wouldn't like to have it reformatted without necessity. 

Also thought about the confirmation step, couldn't agree more, we should describe what the tool is going to do and then ask for the user to proceed or not.

**Pull request merged**